drivers/remote/driver.go

```diff
@@ -247,6 +247,30 @@ func (d *driver) Type() string {
 	return d.networkType
 }
 
+// DiscoverNew is a notification for a new discovery event, such as a new node joining a cluster
+func (d *driver) DiscoverNew(dType driverapi.DiscoveryType, jsonStr string) error {
+	if dType != driverapi.NodeDiscovery {
+		return fmt.Errorf("Unknown discovery type : %v", dType)
+	}
+	notif := &api.DiscoveryNotification{
+		DiscoveryType:     dType,
+		DiscoveryJSONData: jsonStr,
```

Comment:
Why double-encode the JSON, rather than having a field `DiscoveryData: interface{}`? That would make it easier for drivers to decode all at once.